### ISSUE TITLE
company vs company-mode

### ISSUE THREAD

**Issue by guillaumebrunerie** [raxod502/straight.el#917]
## What's wrong

I wanted to try out company-mode, so I ran `straight-use-package 'company-mode`, which installed it, but it was buggy in various ways, including missing icons. After some investigation I found https://github.com/raxod502/straight.el/issues/731 and I realized that the name of the package is `company` and not `company-mode`. So I'm wondering if there is an obsolete `company-mode` recipe lying around somewhere? How do I check where the recipe is coming from?

## Directions to reproduce

From `emacs -Q`, and after running the straight.el bootstrap code (I couldn’t figure out how to use `straight-get-recipe` with `straight-bug-report` as it is an interactive function that doesn’t take the package name as an argument).
```elisp
;; (straight-get-recipe 'company-mode) BUGGY
`(company-mode :type git :host github :repo "company-mode/company-mode" :files (:defaults))

;; (straight-get-recipe 'company) CORRECT
(company :type git :flavor melpa :files (:defaults "icons" "company-pkg.el") :host github :repo "company-mode/company-mode")

;; straight-recipe-repositories
(org-elpa melpa gnu-elpa-mirror el-get emacsmirror-mirror)
```

### Version information

* Emacs version: 27.1
* Operating system: Ubuntu 20.04

**Comment by guillaumebrunerie** [comment]
It looks like it is coming from the `el-get` recipe repository, which is not supposed to be loaded by default according to the doc of `straight.el`. But it does seem to be loaded by default:


<details><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :post-bootstrap 
  (message
   (format "%s" straight-recipe-repositories)))
```
</details>

- Test run at: `2022-01-08 15:55:11`
- system-type: `gnu/linux`
- straight-version: `prerelease (HEAD -> master, origin/master, origin/HEAD) af5437f 2021-09-27`
- emacs-version: `GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0)
 of 2020-09-19`

<details><summary>Output</summary>

```emacs-lisp
Bootstrapping straight.el...
Bootstrapping straight.el...done
Rebuilding all packages due to build cache schema change
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) 92d7daf 2022-01-04
(org-elpa melpa gnu-elpa-mirror el-get emacsmirror-mirror)
Packages:
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master bc3749e3 2022-01-07
"gnu-elpa-mirror"         n/a                  master ebddf26 2021-12-29
"el-get"                  melpa                master 93533097 2021-12-24
"emacsmirror-mirror"      n/a                  master a122c21 2022-01-04
"straight"                n/a                  develop 92d7daf 2022-01-04


```
</details>

**Comment by progfolio** [comment]
> It looks like it is coming from the el-get recipe repository, which is not supposed to be loaded by default according to the doc of straight.el.

Apologies. Seems I forgot to update the docs way back in #549.
I've updated them on the to reflect el-get's inclusion on the develop branch.


Your assessment is correct. MELPA refers to the recipe as `company`, while el-get refers to a slightly different recipe for the same package as `company-mode`. The el-get recipe is missing a `:files` directive. I've submitted a pull to remedy that:

https://github.com/dimitri/el-get/pull/2858


> I couldn’t figure out how to use straight-get-recipe with straight-bug-report as it is an interactive function that doesn’t take the package name as an argument).

You can retrive a recipe programmatically via `straight-recipes-retrieve`:

<details open><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :post-bootstrap 
  (dolist (recipe '(company company-mode))
    (print (straight-recipes-retrieve recipe))))
```
</details>

- Test run at: `2022-01-08 15:04:23`
- system-type: `gnu/linux`
- straight-version: `prerelease (HEAD -> fix/doc-el-get, fork/fix/doc-el-get) 00cd1a6 2022-01-08`
- emacs-version: `GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, X toolkit, cairo version 1.17.4, Xaw3d scroll bars)
 of 2021-12-15`

<details><summary>Output</summary>

```emacs-lisp
Bootstrapping straight.el...
Bootstrapping straight.el...done
Rebuilding all packages due to build cache schema change
Looking for gnu-elpa-mirror recipe -> Cloning melpa...
Looking for gnu-elpa-mirror recipe -> Cloning melpa...done
Looking for emacsmirror-mirror recipe -> Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe -> Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe -> Cloning el-get...
Looking for emacsmirror-mirror recipe -> Cloning el-get...done
Looking for straight recipe -> Cloning emacsmirror-mirror...
Looking for straight recipe -> Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) 3b0f1a1 2022-01-08

(company :type git :flavor melpa :files (:defaults "icons" "company-pkg.el") :host github :repo "company-mode/company-mode")

`(company-mode :type git :host github :repo "company-mode/company-mode" :files (:defaults))

Packages:
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master bc3749e3 2022-01-07
"gnu-elpa-mirror"         n/a                  master ebddf26 2021-12-29
"el-get"                  melpa                master 93533097 2021-12-24
"emacsmirror-mirror"      n/a                  master a122c21 2022-01-04
"straight"                n/a                  develop 3b0f1a1 2022-01-08 


```
</details>

Unfortunately, there's not much we can do on our end to prevent recipe repositories from advertising similar recipes under different names. However, you can specify your preferred sources on globally via `straight-recipe-repositories` and per-recipe via the [`:source`](https://github.com/raxod502/straight.el#the-recipe-format) recipe keyword.

A potential improvement here would be to make it easier for a user to know what source the recipe was returned from.

**Comment by progfolio** [comment]
I'm assuming the information provided addressed the issue. If not, feel free to comment and we can reopen this issue.
